### PR TITLE
AP_GPS: Change search method of GPS type to switch statement.

### PR DESCRIPTION
**Pull request by muramura**
GPS type is retrieved by if sentence.
Therefore,
Shorten the search time by making it switch statement.
Other
Insert white speces and clarify effective range of if statement.

### CODE REVIEW

File: libraries/AP_GPS/AP_GPS.cpp [hunk 1]
```diff
@@ -301,16 +306,25 @@ AP_GPS::detect_instance(uint8_t instance)
...
+    case GPS_TYPE_GSOF:
 		_broadcast_gps_type("GSOF", instance, -1); // baud rate isn't valid
 		new_gps = new AP_GPS_GSOF(*this, state[instance], _port[instance]);
-	} else if ((_type[instance] == GPS_TYPE_NOVA)) {
+        break;
+
+    case GPS_TYPE_NOVA:
 		_broadcast_gps_type("NOVA", instance, -1); // baud rate isn't valid
 		new_gps = new AP_GPS_NOVA(*this, state[instance], _port[instance]);
+        break;
```
Comment (OXINARF) [inline libraries/AP_GPS/AP_GPS.cpp]: All the breaks aren't aligned with previous lines (because those lines have tabs instead of spaces).

**Comment by OXINARF** [comment]
@WickedShell how do you feel about this?

**Comment by muramura** [comment]
The tab code was corrected to White spaces.

**Comment by rmackay9** [comment]
I think this makes the code easier to read.  I vote for it going in.

**Comment by lvale** [comment]
@muramura The code reads much better and easier to understand. One question that I have is which version of the code, ie with IF's or Case/Switch is more efficient in terms of compiled code, processor usage, execution time?

**Comment by muramura** [comment]
@lvale Comments thank you.
It means that it takes time to search.
It does not measure.

I would like to measure the time from the beginning of the search until the last hit.
Time uses the micros () method.

**Comment by magicrub** [comment]
Looks good to me

**Comment by muramura** [comment]
@lvale , @magicrub Comment Thank you.

**result
It is faster to switch to a sentence than to search with multiple if statements.**

Test contents
Measure the time with the if statement and switch statement on the GPS type search.
GPS type is "GPS_TYPE_NOVA".
Hardware is IRIS.
State is DISARMED
(Copter V3.5-dev (c846cc24)): If statement is **0, 1, 8 or 9 micro seconds**.
(Copter V3.5-dev (c022fd5e)): Switch statement is **0, 1 micro seconds**.

test code
===

Case if statement

    _type[instance] = GPS_TYPE_NOVA;   // CHECK TIME
    uint64_t uss = AP_HAL::micros64(); // CHECK TIME

    if (_type[instance] == GPS_TYPE_PX4) {
:
:
	} else if ((_type[instance] == GPS_TYPE_NOVA)) {
        uint64_t use = AP_HAL::micros64();
        GCS_MAVLINK::send_statustext_all(MAV_SEVERITY_CRITICAL,"###us=%lld", use - uss);

＝＝＝
Case switch statement

    _type[instance] = GPS_TYPE_NOVA;   // CHECK TIME
    uint64_t uss = AP_HAL::micros64(); // CHECK TIME

    switch (_type[instance]) {
#if CONFIG_HAL_BOARD == HAL_BOARD_PX4
：
：
   case GPS_TYPE_NOVA:
        uint64_t use = AP_HAL::micros64();
        GCS_MAVLINK::send_statustext_all(MAV_SEVERITY_CRITICAL,"###us=%lld", use - uss);
＝＝＝
message log 

 if statement
 
 GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
**###us=8**
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
**###us=9**
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
**###us=9**
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
PX4v2 004A0039 30345111 37383933
Frame: QUAD
PX4: 96a4c296 NuttX: 580f5354
APM:Copter V3.5-dev (c846cc24)
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1

-----------------------

case statement

GPS 0: detected as NOVA
###us=0
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=1
GPS 0: detected as NOVA
###us=0
PX4v2 004A0039 30345111 37383933
Frame: QUAD
PX4: 96a4c296 NuttX: 580f5354
APM:Copter V3.5-dev (c022fd5e)
GPS 0: detected as NOVA
###us=0
PreArm: Throttle below Failsafe
PreArm: Accels not calibrated
PreArm: check fence
PreArm: Need 3D Fix
PreArm: Compass not calibrated

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**